schutzbot/scheduled_cloud_cleaner.sh

```diff
@@ -179,11 +179,33 @@ for snapshot in ${SNAPSHOTS}; do
 	START_TIME=$(echo "${snapshot}" | jq '.StartTime' | tr -d '"')
 
 	if [[ ${TAG} == "gitlab-ci-test" && ${TAG_VALUE} == "true" ]]; then
-		if [[ $(date -d "${START_TIME}" +%s) -lt $(date -d "- 2 minutes" +%s) ]]; then
+		if [[ $(date -d "${START_TIME}" +%s) -lt "${DELETE_TIME}" ]]; then
 			$AWS_CMD ec2 delete-snapshot --snapshot-id "${SNAPSHOT_ID}"
 			echo "The snapshot with id ${SNAPSHOT_ID} was deleted"
 		else
 			echo "The snapshot with id ${SNAPSHOT_ID} was created less than ${HOURS_BACK} hours ago"
 		fi
 	fi
 done
+
+# Remove tagged and old enough s3 objects
+AWS_BUCKET=composerredhat
+OBJECTS=$($AWS_CMD s3api list-objects --bucket ${AWS_BUCKET} | jq -c .Contents[])
+
+for object in ${OBJECTS}; do
+        LAST_MODIFIED=$(echo "${object}" | jq '.LastModified' | tr -d '"')
+        OBJECT_KEY=$(echo "${object}" | jq '.Key' | tr -d '"')
+
+        if [[ $(date -d "${LAST_MODIFIED}" +%s) -lt ${DELETE_TIME} ]]; then
+                TAG=$($AWS_CMD s3api get-object-tagging --bucket "${AWS_BUCKET}" --key "${OBJECT_KEY}" | jq .TagSet[0].Key | tr -d '"')
+                TAG_VALUE=$($AWS_CMD s3api get-object-tagging --bucket "${AWS_BUCKET}" --key "${OBJECT_KEY}" | jq .TagSet[0].Value | tr -d '"')
+
+                if [[ ${TAG} == "gitlab-ci-test" && ${TAG_VALUE} == "true" ]]; then
+                        $AWS_CMD s3 rm "s3://${AWS_BUCKET}/${OBJECT_KEY}"
+                        echo "The object with key ${OBJECT_KEY} was removed"
+                else
+                        echo "The object with key ${OBJECT_KEY} was edited less than ${HOURS_BACK} hours ago"
```

Comment:
I would just remove this `else` statement. You're already filtering out anything that was modified less than `$HOURS_BACK` ago in the previous `if` on line 199. And we don't have to mention every untagged object...

Reply:
I didn't thought about this. I doesn't make sense to mention every untagged object. Thanks!